packages/lodestar-beacon-state-transition/src/fast/block/processRandao.ts

```diff
@@ -16,23 +15,39 @@ export function processRandao(
   const config = epochCtx.config;
   const epoch = epochCtx.currentShuffling.epoch;
   const randaoReveal = body.randaoReveal.valueOf() as Uint8Array;
+
   // verify RANDAO reveal
   if (verifySignature) {
-    const proposerIndex = epochCtx.getBeaconProposer(state.slot);
-    const proposerPubkey = epochCtx.index2pubkey[proposerIndex];
-    const signingRoot = computeSigningRoot(
-      config,
-      config.types.Epoch,
-      epoch,
-      getDomain(config, state, DomainType.RANDAO)
-    );
-    if (!bls.Signature.fromBytes(randaoReveal).verify(proposerPubkey, signingRoot)) {
+    const signatureSet = getRandaoRevealSignatureSet(epochCtx, state, body);
+    if (!verifySignatureSet(signatureSet)) {
       throw new Error("RANDAO reveal is an invalid signature");
     }
   }
+
   // mix in RANDAO reveal
   state.randaoMixes[epoch % config.params.EPOCHS_PER_HISTORICAL_VECTOR] = xor(
     Buffer.from(getRandaoMix(config, state, epoch) as Uint8Array),
     Buffer.from(hash(randaoReveal))
   );
 }
+
+/**
+ * Extract signatures to allow validating all block signatures at once
+ */
+export function getRandaoRevealSignatureSet(
+  epochCtx: EpochContext,
+  state: BeaconState,
+  body: BeaconBlockBody
+): ISignatureSet {
+  const config = epochCtx.config;
+  const epoch = epochCtx.currentShuffling.epoch;
+  const proposerIndex = epochCtx.getBeaconProposer(state.slot);
```

Comment:
should pass in the whole beacon block, so we can do `epochCtx.getBeaconProposer(block.slot);` and avoid needing the state dialed per-slot.

See https://github.com/sigp/lighthouse/blob/c6baa0eed131c5e8ecc5860778ffc7d4a4c18d2d/consensus/state_processing/src/per_block_processing/signature_sets.rs#L121